### PR TITLE
Testing Enumerable with Rspec

### PR DESCRIPTION
**Pull request by cgcarlosg**
Create and include files for appropriate Rspec testing
Use of Rspec to test public functions of custom classes

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  describe '#my_select' do
+    it 'return a new array based on block provided' do
+      expect(array.my_select(&:positive?)).to eql([1, 2, 3])
+    end
+
+    it 'return a new hash based on block provided' do
+      expect(hash.my_select { |_key, value| value == 'enum' }).to eql([%w[a enum]])
+    end
+  end
```
Comment (ghost) [inline spec/enumerable_spec.rb]: In `#my_select `method. You have written tests for the correct usage and positive scenario of the method which is, `it 'returns an array based on block provided'`. Now write a test for a negative scenario, which can be something like, `it 'does not return an array when no block is provided'`. Do the same for the hash check. See [screenshot ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from pathwright.

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    it 'return true if all elements meet block given' do
+      expect(array.my_all? { |num| num == 1 }).to eql(false)
+      expect(hash.my_all? { |_key, value| value == 'enum' }).to eql(false)
+    end
+
+    it 'return false if one element does not meet block given' do
+      expect(array.my_all? { |num| num.is_a? Numeric }).to eql(true)
+      expect(hash.my_all? { |value| value.is_a? Enumerable }).to eql(true)
+    end
+  end
```
Comment (ghost) [inline spec/enumerable_spec.rb]: Cover the scenario for when no block is given.
The expected behaviour for each `it` block is wrongly described in the` it` statement. The first `it` block statement says the expected results should be true values, yet the results of the tests are false values. The second statement says the expected result is a false value yet the results of the tests are true values.

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  describe '#my_map' do
+    it 'return new array with elements modified by block' do
+      expect(array.my_map { |num| num * 2 }).to eql([2, 4, 6])
+    end
+    it 'return new hash with elements modified by block' do
+      expect(array.my_map { 'hello' }).to eql(%w[hello hello hello])
+    end
+  end
```
Comment (ghost) [inline spec/enumerable_spec.rb]: Cover the negative scenario for when no block is given.

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  describe '#my_each_with_index' do
+    it 'returns a value from an index in an array' do
+      i = 0
+      array.my_each_with_index do |value, index|
+        expect(index).to eql(i)
+        expect(value).to eql(array[i])
+        i += 1
+      end
+    end
+  end
```
Comment (ghost) [inline spec/enumerable_spec.rb]: Specs should cover positive and negative scenarios. In both `my_each` and `my_each_with_index`, the positive scenarios are when a block is given. Cover the scenarios for when a block is not given for each of these methods. See [screenshot ](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from pathwright.

File: spec/enumerable_spec.rb [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    it 'return true if no element meets block given' do
+      expect(array.my_none? { |num| num == 0 }).to eql(true)
+      expect(hash.my_none? { |_key, value| value == 0 }).to eql(true)
+    end
+
+    it 'return false if any element meets block given' do
+      expect(array.my_none? { |num| num == 1 }).to eql(false)
+      expect(hash.my_none? { |_key, value| value == 'enum' }).to eql(false)
+    end
+  end
```
Comment (ghost) [inline spec/enumerable_spec.rb]: Cover the scenario for when no block is given for `my_none?` and `my_any?` methods.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required

## Hello team 👋.

**Great job on your project 💪. Your tests are well written 👌.**

**There are a few fixes required, however, before the project can be approved.**

1). **In your readme, the run test manual says that I should run `rspec +` for spec tests. When I run that it throws an error. Also, the statement above `rspec +` has a typo. It should be directory of not directory od.**

2). **Check the comments below.**

### Happy coding 😃.

**PS: You can reach me on slack (Rose Wanjohi).**

**Review by Temmarie — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @cgcarlosg and @Janluc ! 👋 👋 

### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:

### You may merge, good luck with your next projects! :rocket:
### Happy Coding! :+1:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Comment by cgcarlosg** [comment]
Project done!

**Pull request closed without merging**